overlord/ifacestate/hotplug_test.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ifacestate_test
+
+import (
+	"github.com/snapcore/snapd/interfaces/hotplug"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+
+	. "gopkg.in/check.v1"
+)
+
+type hotplugSuite struct{}
+
+var _ = Suite(&hotplugSuite{})
+
+func (s *hotplugSuite) TestDefaultDeviceKey(c *C) {
+	di, err := hotplug.NewHotplugDeviceInfo(map[string]string{
```

Comment:
(drive-by) I would love to see a test that uses a copy of real device data to see what a key looks like in practice.

Reply:
Added 3 samples made of real devices. Note, these examples needs to be taken a with a grain of salt... it's a little bit of an uncharted territory, there may be devices which need special treatment (dedicated implementation of HotplugDeviceKey method on the interface, or extra attributes added to this defaultDeviceKey helper).